client/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,151 @@
+import { cn } from '@/lib/utils';
+import { Button } from '@/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { Link, useNavigate } from 'react-router-dom';
+import { FormEvent } from 'react';
+import { useState } from 'react';
+import { User, useUser } from '@/components/useUser';
+
+type AuthData = {
+  user: User;
+  token: string;
+};
+
+export default function LoginForm({
+  className,
+  ...props
+}: React.ComponentPropsWithoutRef<'div'>) {
+  const [isLoading, setIsLoading] = useState(false);
+  const navigate = useNavigate();
+  const { handleSignIn } = useUser();
+
+  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      const formData = new FormData(event.currentTarget);
+      const userData = Object.fromEntries(formData);
+      const req = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(userData),
+      };
+      const res = await fetch('/api/auth/register', req);
+      if (!res.ok) {
+        throw new Error(`fetch Error ${res.status}`);
+      }
+      const data = await res.json();
+      const { user } = data;
+      alert(
+        `Successfully registered ${user.userName} as userId ${user.userId}.`
+      );
+      navigate('/dashboard');
+    } catch (err) {
+      alert(`Error registering user: ${err}`);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function handleGuestLogin() {
```

Comment:
This looks like a duplicate of the same function in LoginPage. Extract it (or, at least the fetch call, etc.) to a utility function that can be shared.